bin/anthology/venues.py

```diff
@@ -121,18 +121,25 @@ def load_from_dir(self, directory):
 
     def get_by_letter(self, letter):
         """Get a venue acronym by first letter (e.g., Q -> TACL)."""
-        try:
-            return self.letters[letter]
-        except KeyError:
-            log.critical("Unknown venue letter: {}".format(letter))
+        return self.letters.get(letter, None)
 
     def get_main_venue(self, anthology_id):
         """Get a venue acronym by anthology ID (e.g., acl -> ACL)."""
-        collection_id, *_ = deconstruct_anthology_id(anthology_id)
+        collection_id, volume_id, _ = deconstruct_anthology_id(anthology_id)
         if is_newstyle_id(collection_id):
             return self.acronyms_by_key[collection_id.split(".")[-1]]
-        else:
-            return self.get_by_letter(collection_id[0])
+        else:  # old-style ID
+            main_venue = self.get_by_letter(collection_id[0])
+            if main_venue is None:
+                try:
+                    main_venue = self.joint_map[
+                        build_anthology_id(collection_id, volume_id, None)
+                    ][0]
```

Comment:
Note to self: maybe also error out if there's more than one venue in this list, in the unlikely case this ever pops up in the future and confuses people.